pkg/operator/operator.go

```diff
@@ -606,3 +588,63 @@ func (o *Operator) setMutatingWebhookCABundle(ctx context.Context, caBundle []by
 	}
 	return o.client.Update(ctx, &mwc)
 }
+
+func (o *Operator) canUpdateWebhooks(ctx context.Context) (bool, error) {
+	return o.canI(ctx,
+		authv1.ResourceAttributes{
+			Group:     arv1.GroupName,
+			Resource:  "MutatingWebhookConfiguration",
+			Namespace: o.opts.OperatorNamespace,
+			Verb:      "update",
+		},
+		authv1.ResourceAttributes{
+			Group:     arv1.GroupName,
+			Resource:  "ValidatingWebhookConfiguration",
+			Namespace: o.opts.OperatorNamespace,
+			Verb:      "update",
+		},
+	)
+}
+
+func (o *Operator) canI(ctx context.Context, resources ...authv1.ResourceAttributes) (bool, error) {
+	for _, resource := range resources {
+		req := authv1.SelfSubjectAccessReview{
+			Spec: authv1.SelfSubjectAccessReviewSpec{
+				ResourceAttributes: &resource,
+			},
+		}
+		if err := o.client.Create(ctx, &req); err != nil {
```

Comment:
**thought (non-blocking):** This is cool! I didn't realize that you could check permissions this way. Do you know how this compares to something like a dry-run client?

Reply:
`can-i` just checks for permissions without checking for any other kind of API conformity, like a dry-run client might.